sno/working_copy/postgis.py

```diff
@@ -116,83 +119,113 @@ def session(self, bulk=0):
             finally:
                 self._db.close()
                 del self._db
-                L.debug(f"session: new/done")
+                L.debug("session: new/done")
+
+    _V2_TYPE_TO_PG_TYPE = {
+        "boolean": "boolean",
+        "blob": "bytea",
+        "date": "date",
+        "float": {0: "real", 32: "real", 64: "double precision"},
+        "geometry": "geometry",
+        "integer": {
+            0: "integer",
+            16: "smallint",
+            32: "integer",
+            64: "bigint",
+        },
+        "interval": "interval",
+        "numeric": "numeric",
+        "text": "text",
+        "time": "time",
+        "timestamp": "timestamp",
```

Comment:
what happened here wrt time zones?

Reply:
Not very sure. Getting this perfectly right will take some more thought - I don't know if we have a plan for what to do when working copies support different things eg one working copy supports "with timezone" and "without timezone" (ie POSTGIS) variants, the other does not (GPKG). I wrote the mapping above because its simple - I'll add a TODO to revisit this, but I can't write the final revision of this code without a plan for how we roundtrip types through less full-featured working copies.

Reply:
Yeah, needs a separate discussion about the general approach, then details for each thing we run into.